pandas/core/generic.py

```diff
@@ -9730,14 +9731,13 @@ def _where(
                     raise InvalidIndexError
 
                 elif other.ndim < self.ndim:
-                    # TODO(EA2D): avoid object-dtype cast in EA case GH#38729
                     other = other._values
                     if axis == 0:
                         other = np.reshape(other, (-1, 1))
+                        other = tile_compat(other, self.shape[1], axis=1)
                     elif axis == 1:
                         other = np.reshape(other, (1, -1))
-
-                    other = np.broadcast_to(other, self.shape)
```

Comment:
the point of this is that other.shape will match self.shape.  IIUC tile_compat won't accomplish this

Reply:
I think it should be fine here:
- `other` is originally 1D, and is the same length as `self` (or the safe width if `axis=1`)
- it's then made 2D with in the line just above this one (`other = np.reshape(other, (1, -1))` or `other = np.reshape(other, (-1, 1))`)
- it then needs expanding so its width matches the width of self (resp. length if `axis=1`). Both `np.broadcast_to` and `tile_compat` will do this:

```
(Pdb) p other
<IntegerArray>
[
[7],
[8],
[<NA>]
]
Shape: (3, 1), dtype: Int64
(Pdb) tile_compat(other, self.shape[1], axis=1)
<IntegerArray>
[
[7, 7],
[8, 8],
[<NA>, <NA>]
]
Shape: (3, 2), dtype: Int64
(Pdb) np.broadcast_to(other, self.shape)
array([[7, 7],
       [8, 8],
       [<NA>, <NA>]], dtype=object)
```

They're the same shape, but one is still `IntegerArray` whilst the other one is and object `np.array`

Reply:
i think that works for IntegerArray because it has partial 2D support (i started it but never really went anywhere), but will fail with e.g. IntervalArray

Reply:
sure but then wouldn't it already fail on `.reshape(-1, 1)` (or `.reshape(1, -1)`) from L9738?

Reply:
ah no, that does indeed work
```
In [13]: np.reshape(pd.arrays.IntervalArray([pd.Interval(0, 1), pd.Interval(1, 5)]), (-1, 1))
Out[13]: 
array([[Interval(0, 1, closed='right')],
       [Interval(1, 5, closed='right')]], dtype=object)
```

closing then as this isn't the solution, thanks for catching this!

Reply:
in the absence of 2D EA support, i expect the solution will be a bit like in DataFrame.astype:

```
if other.ndim < self.ndim:
    values = extract_array(values)
    if is_1d_only_ea_object(values):
          if axis == 0:
                 sers = [self.iloc[:, i]._where(...)] for i in range(self.shape[1])
                 ...
           else: ...
```